docs/wiki/01_STUDY_PARAMETERS_SETUP/02_Clinical_Instruments/02_instrument_format/PHP_instrument.md

```diff
@@ -0,0 +1,220 @@
+- In the `_setupForm()` function the keyword `<TEST_NAME>` in the `preg_match()` function should be replaced by the `TEST_NAME` value.
+- Many other Configuration can alter the behaviour of the instrument and automated quality assurance that is undergoes. Refer to the [Additional Configurations](# Additional Configurations) section below for more information.
+
+Finally, the `getFullName()` function must be changed to reflect the correct full name of the instrument.
+
+#### Pages
+
+Instrument can be broken down into pages. An instrument has a minimum of one page and no maximum number of pages. Pages can have customized names and are configured in the instrument's `getSubtestList()` function. 
+
+The instrument class' `_main()` function represents the first page, also referred to as the "Top" page. All subsequent pages are commonly implemented as `_page1()`, `_page2()`, ... Although this is the most common use case, it is possible to name the functions differently as long as all other configurations are modified accordingly.
+
+In the `getSubtestList()`, make sure to properly refactor the `Name` key of each element in the array to match the following convention `<TEST_NAME><PAGE_FUNCTION_NAME>` where `<Test_NAME>` is replaced by the `TEST_NAME` value you have chosen and `<PAGE_FUNCTION_NAME>` is replaced by `_page1`, `_page2` or the appropriate value if you elected to rename the page functions.
+
+If you chose to rename the page functions (differently than sequential `_pageX`) you must modify the `_setupForm()` function to make sure the `preg_match()` call is properly configured to match all your different pages. This `preg_match()` allows for proper routing to the pages of the instrument.
+
+#### Form Elements
+
+As mentioned above, inheritance from the `NDB_BVL_Instrument.class.inc` enables the instrument under development to use several pre-implemented form wrappers to generate all sorts of fields. The full list of usable wrappers spans across the classes `php/libraries/NDB_BVL_Instrument.class.inc`, `php/libraries/NDB_Page.class.inc` and `php/libraries/LorisForm.class.inc`. 
+
+As a general rule, functions starting with `add...()` create and append the element to the form directly; functions starting with `create...()` only create the form element. The latter use case is necessary for grouping. Groups are either a multi-element line in the instrument or tables where each row of the table is a group. Examples of groups, tables and singular form elements can be found in the `raisinbread/instruments/` directory.
+
+Below are a few best practices for cleaner functional forms:
+
+- Element names must be lowercase and fewer than 64 characters (e.g. `q07_mother_maiden_name`). Avoid using special characters other than underscores(`_`).
+- Element names `*_status` are reserved for select boxes accompanying text fields.
```

Comment:
It might be worth noting here that if an `x_status` field is set, then `x` automatically has its data wiped out on save..